Source/ProgressionSystemRuntime/Private/Components/PSSpotComponent.cpp

```diff
@@ -47,6 +47,50 @@ void UPSSpotComponent::OnReset_Implementation()
 	}
 }
 
+// Listen game states to switch character skin. 
+void UPSSpotComponent::OnGameStateChanged_Implementation(ECurrentGameState CurrentGameState)
+{
+	// it's possible that spot might not be loaded till that time so no ensure added
+	UPSSpotComponent* CurrentSpot = UPSWorldSubsystem::Get().GetCurrentSpot();
+
+	// check the spot  
+	if (CurrentSpot == nullptr || CurrentSpot != this)
+	{
+		return;
+	}
+
+	if (CurrentGameState == ECurrentGameState::GameStarting)
+	{
+		UMySkeletalMeshComponent& MeshComp = GetMeshChecked();
+		const int32 CurrentSkinIndex = MeshComp.GetAppliedSkinIndex();
+		bool isCurrentSkinAvailable = MeshComp.IsSkinAvailable(CurrentSkinIndex);
```

Comment:
please follow Unreal naming standards: any boolean should start from `b`

Reply:
done https://github.com/moinm3uw/ProgressionSystem/pull/63/commits/aca6b68a44acab34497a7c83e3d0e17680a3c834